runtime/src/accounts_db.rs

```diff
@@ -2970,7 +2972,7 @@ impl AccountsDb {
                 .map(|dirty_store_chunk| {
                     let mut oldest_dirty_slot = max_slot_inclusive.saturating_add(1);
                     dirty_store_chunk.iter().for_each(|(slot, store)| {
-                        if is_ancient(&store.accounts) {
+                        if slot < &oldest_non_ancient_slot {
```

Comment:
👍 

If we will use this check often, maybe consider making it a function like `is_ancient` but takes slot instead.